src/plugins/intel_gpu/src/plugin/ops/bucketize.cpp

```diff
@@ -15,10 +15,18 @@ namespace {
 void CreateBucketizeOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v3::Bucketize>& op) {
     validate_inputs_count(op, {2});
 
+    cldnn::bucketize_boundary boundary_status = cldnn::bucketize_boundary::dynamic;
+    if (op->get_input_partial_shape(1).is_static()) {
+        boundary_status = shape_size(op->get_input_shape(1)) == 0 ? cldnn::bucketize_boundary::empty
```

Comment:
I still don't see why you need any special attribute here. You have shape (either dynamic or static) during kernel code jitting, so you can check if it's dynamic there.

Reply:
Actually, this special attribute is required for static case with an empty second input.

For static empty second input, after _CreateBucketizeOp()_, the shape becomes [1] and the value is 0 while the expected shape is [0]. That means, for static case, empty shape does not remain empty after _CreateBucketizeOp()_. On the other hand, for dynamic shape, it works as expected.

Thus, for static case, whether the second input is empty or not has to be determined in _CreateBucketizeOp()_, but for dynamic case, it has to be determined at execution time.

That' why this attribute is required.

Reply:
`CreateBucketizeOp` doesn't modify shapes anyhow, so I believe the issue is somewhere else. Please remove this attribute and handle shape issues in the shape inference part or in the kernel itself.

Reply:
"after `CreateBucketizeOp`, the shape becomes [1] and the value is 0" doesn't mean that `CreateBucketizeOp` modify shape. At the stage of the shape inference or kernel, the empty second input is already filled with 0. Even in the time of initializing the graph, the empty second input is already filled with 0.

Reply:
> the shape becomes [1] and the value is 0 while the expected shape is [0]
> At the stage of the shape inference or kernel, the empty second input is already filled with 0


That sounds like shape inference issue. It shouldn't be resolved by adding new attributes for the primitive.